### ISSUE TITLE
How many epochs are used for each run in Table 3?

### ISSUE THREAD

**Issue by Gengsheng-Li** [lasgroup/SDPO#17]
Hi,
Thanks for your previous help!
I am trying to reproduce the Chemistry results reported in Table 3 (Section 3), but I noticed a significant discrepancy in training time. The paper mentions that each run takes approximately 6 hours on 4 GH200 GPUs. However, with the default configuration in this repository (total_epochs: 30), the Chemistry dataset (2,160 training samples, train_batch_size=32) results in approximately 2,010 training steps, which takes around 70 hours on my setup.
Even accounting for hardware differences, this gap seems too large to be explained by compute alone. I don't know if the experiments in Table 3 used a different training configuration or there is something wrong with my training setup.

My setup:
8 × NVIDIA H20 GPUs (96GB HBM3 each)
FlashAttention enabled
SGLang as the rollout engine
Conda environment installed following the latest setup guide

Thank you for your time! :-D

**Comment by jonhue** [comment]
Hi. Yes, the training time with 30 epochs is substantially longer. In the paper, we only report results until 5h training (time measured based on cumulative `timing_s/step`). We never actually ran it for 30 epochs.

**Comment by Gengsheng-Li** [comment]
@jonhue Many thanks! Since we are using different hardware setups, could you let us know roughly how many training steps you had completed by around the 5-hour mark (e.g., on the Chemistry benchmark, or whichever is most convenient for you)? This would help us establish a more objective basis for reproduction. :-D

**Comment by jonhue** [comment]
I think it was around 150 steps.

**Comment by Gengsheng-Li** [comment]
@jonhue Thanks for the clarification! I'd like to confirm — is val-aux/sciknoweval/score/mean@16 the metric reported in Table 3 of the paper?
In our Chemistry experiment, the score reaches around 63 at step 150, and the highest score we observed is approximately 0.69 at step 270. However, Table 3 in the paper reports a score of 70 in 5 hour, i.e., maybe around 150 steps. Could you help clarify what might account for this discrepancy? Are there any additional hyperparameter settings or evaluation details we might be missing? :-D

<img width="576" height="392" alt="Image" src="https://github.com/user-attachments/assets/07e31210-f3bd-48b7-80ae-658894aaeaf3" />

<img width="577" height="394" alt="Image" src="https://github.com/user-attachments/assets/7ef6e0df-cc00-4c8a-81a3-75bee6ecbde9" />

**Comment by jonhue** [comment]
Hi @Gengsheng-Li , sorry for my late reply!

We recently reran the Chemistry experiment with this latest code. We reach ~0.8 with Olmo and 0.7 with Qwen (just a little bit higher than what you get).
In https://github.com/lasgroup/SDPO/pull/20, we also uploaded our datasets as JSON.
Hope this helps!

**Comment by Gengsheng-Li** [comment]
@jonhue 
Hi, thank you for your reply, I will look throught the updates! :-D

**Comment by nasosger** [comment]
@jonhue May I ask some follow-up (related) questions on this? 
- In your updated paper, you mention ~0.8 on Chemistry with Qwen3-8B. Could this be a typo? (just asking since here you mention ~0.7).
- In these runs (generalization), are you using fixed random seeds?
- In the updated results, I see that some of the reported scores are different in chemistry and biology benchmarks. Could this be attributed to different training-validation splits, or have you changed any hyperparameter on either training or eval? 

Thank you for sharing the codebase! Also tagging @Gengsheng-Li (OP) in case you have any updates on reproducing these runs.

**Comment by jonhue** [comment]
@nasosger sure :)

* It's 0.8 as also seen in the W&B logs: https://wandb.ai/jonhue/SDPO?nw=mgotcx6kk7
* Unfortunately, vLLM is not possible to seed afaik. However, we repeat each run 3 times with relatively minor differences. The update paper reports mean and error bars. Our data processing is seeded.
* We slightly changed the training/validation datasets to explicitly remove any non multiple-choice questions. This changed the absolute numbers, but didn't really change how they compare relatively.

**Comment by nasosger** [comment]
@jonhue  thanks for the reply :)
- Indeed its 0.8, I was just using the unfiltered dataset (with the open-ended questions as well). Sorry about that! 
- As for vllm, I think verl uses a default seeding when initializing the engine but there is randomness when generating the rollouts.

Anyway thanks for the help and congrats for your awesome work!

**Comment by kim0806** [comment]
Hi authors,

I tried to reproduce the results on the chemistry dataset（qwen3-8b）, but I was unable to obtain the conclusions reported in the paper. Specifically, after around 100 training steps, the validation mean@16 starts to gradually decrease.

I used the same hyperparameters as those provided in the appendix and ran the experiments on 4 H100 GPUs. The only difference is that my inference engine is SGLang.

Could this discrepancy be caused by the inference engine, or are there any other implementation details I might have missed?

Thank you for your help!

<img width="1782" height="878" alt="Image" src="https://github.com/user-attachments/assets/f49428c7-90f0-46ed-9960-95a6073766db" />

<img width="1780" height="890" alt="Image" src="https://github.com/user-attachments/assets/43237025-b68d-4e56-8b67-93128e7c7b9a" />
